proposals/lip-0054.md

```diff
@@ -331,15 +351,27 @@ def verify(trs: Transaction) -> None:
 
     if terminatedOutboxAccount(trsParams.chainID) does not exist:
         raise Exception("Terminated outbox account does not exist.")
+    
+    # Check that there is at least one cross-chain message to recover.
+    if len(trsParams.crossChainMessages) == 0:
+        raise Exception("No cross-chain messages to recover.")
+    
+    # Check that there is exactly one index per cross-chain message.
+    if len(trsParams.idxs) != len(trsParams.crossChainMessages):
+        raise Exception("Inclusion proof indices and cross-chain messages do not have the same length.")
 
     # Check that the idxs are sorted in ascending order.
     if trsParams.idxs != sorted(trsParams.idxs):
         raise Exception("Cross-chain message indexes are not sorted in ascending order.")
 
-    # Check that the CCMs are still pending.
-    for index in trsParams.idxs:
-        if index < terminatedOutboxAccount(trsParams.chainID).partnerChainInboxSize:
-            raise Exception("Cross-chain message is not pending.")
+    # Check that the idxs are unique.
+    if len(trsParams.idxs) != len(set(trsParams.idxs)):
+        raise Exception("Cross-chain message indexes are not unique.")
```

Comment:
This is a good point. Due to this, I think we also need to ensure in the state recover transactions that the `(substorePrefix, storeKey)` pairs are pair-wise distinct. Otherwise, it seems, for instance, possible to recover a token account several times in one transaction.

Reply:
I added this check in the verification